numpy/_core/src/umath/loops_unary_fp.dispatch.cpp

```diff
@@ -0,0 +1,219 @@
+#define _UMATHMODULE
+#define _MULTIARRAYMODULE
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#include "numpy/ndarraytypes.h"
+#include "numpy/npy_common.h"
+#include "numpy/npy_math.h"
+#include "numpy/utils.h"
+#include "fast_loop_macros.h"
+#include "loops_utils.h"
+#include <hwy/highway.h>
+#include <hwy/aligned_allocator.h>
+
+namespace hn = hwy::HWY_NAMESPACE;
+
+// Alternative to per-function HWY_ATTR: see HWY_BEFORE_NAMESPACE
+#define SUPER(NAME, FUNC, IS_RECIP)                                            \
```

Comment:
Please refrain from using macro's for functions these large. They are hard to read and will be a pain to debug. Could we make use of templates here?

Reply:
> Please refrain from using macro's for functions this large. They are hard to read and will be a pain to debug. Could we make use of templates here?

I try to use lama func but occurs a error:
```
(num) luyahan@plct-c7:~/source/numpy$ spin build 
$ /home/luyahan/source/num/bin/python vendored-meson/meson/meson.py compile -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /home/luyahan/source/num/bin/ninja -C /home/luyahan/source/numpy/build
ninja: Entering directory `/home/luyahan/source/numpy/build'
[8/35] Generating numpy/generate-version with a custom command
Saving version to numpy/version.py
[12/35] Compiling C++ object numpy/_core/libloops_unary_fp.dispatch.h_SSE42.a.p/src_umath_loops_unary_fp.dispatch.cpp.o
FAILED: numpy/_core/libloops_unary_fp.dispatch.h_SSE42.a.p/src_umath_loops_unary_fp.dispatch.cpp.o 
c++ -Inumpy/_core/libloops_unary_fp.dispatch.h_SSE42.a.p -Inumpy/_core -I../numpy/_core -Inumpy/_core/include -I../numpy/_core/include -I../numpy/_core/src/common -I../numpy/_core/src/multiarray -I../numpy/_core/src/npymath -I../numpy/_core/src/umath -I../numpy/_core/src/highway -I/usr/include/python3.10 -I/usr/include/x86_64-linux-gnu/python3.10 -I/home/luyahan/source/numpy/build/meson_cpu -fdiagnostics-color=always -Wall -Winvalid-pch -std=c++17 -O2 -g -msse -msse2 -msse3 -fPIC -DNPY_INTERNAL_BUILD -DHAVE_NPY_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -D__STDC_VERSION__=0 -fno-exceptions -fno-rtti -O3 -DNPY_HAVE_SSE2 -DNPY_HAVE_SSE -DNPY_HAVE_SSE3 -DNPY_HAVE_SSSE3 -DNPY_HAVE_SSE41 -DNPY_HAVE_POPCNT -DNPY_HAVE_SSE42 -msse -msse2 -msse3 -mssse3 -DHWY_WANT_SSSE3=1 -msse4.1 -mpopcnt -msse4.2 -DHWY_WANT_SSE4=1 -DNPY_MTARGETS_CURRENT=SSE42 -MD -MQ numpy/_core/libloops_unary_fp.dispatch.h_SSE42.a.p/src_umath_loops_unary_fp.dispatch.cpp.o -MF numpy/_core/libloops_unary_fp.dispatch.h_SSE42.a.p/src_umath_loops_unary_fp.dispatch.cpp.o.d -o numpy/_core/libloops_unary_fp.dispatch.h_SSE42.a.p/src_umath_loops_unary_fp.dispatch.cpp.o -c ../numpy/_core/src/umath/loops_unary_fp.dispatch.cpp
In file included from ../numpy/_core/src/highway/hwy/highway.h:482,
                 from ../numpy/_core/src/umath/loops_unary_fp.dispatch.cpp:14:
../numpy/_core/src/highway/hwy/ops/x86_128-inl.h: In function ‘vec_f64 Round(vec_f64)’:
../numpy/_core/src/highway/hwy/ops/x86_128-inl.h:11224:27: error: inlining failed in call to ‘always_inline’ ‘hwy::N_SSE4::Vec128<double, N> hwy::N_SSE4::Round(hwy::N_SSE4::Vec128<double, N>) [with long unsigned int N = 2]’: target specific option mismatch
11224 | HWY_API Vec128<double, N> Round(const Vec128<double, N> v) {
      |                           ^~~~~
../numpy/_core/src/umath/loops_unary_fp.dispatch.cpp:115:51: note: called from here
  115 | static vec_f64 Round(vec_f64 x) { return hn::Round(x); }
```

Reply:
https://github.com/numpy/numpy/pull/21057/files#diff-215e636f5e9bff97bb376cc620e8d7941dc75facd96a8b114b1908bbe355e31a might serve as a good example to follow.